hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hss/signschedule/SignScheduleTranslator.java

```diff
@@ -16,32 +16,171 @@
+import static com.hedera.node.app.service.contract.impl.utils.ConversionUtils.maybeMissingNumberOf;
+import static com.hedera.node.app.service.contract.impl.utils.ConversionUtils.numberOfLongZero;
+import static com.hedera.node.app.spi.workflows.HandleException.validateTrue;
+import static java.util.Collections.emptySet;
 import static java.util.Objects.requireNonNull;
 
+import com.esaulpaugh.headlong.abi.Address;
 import com.esaulpaugh.headlong.abi.Function;
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.ContractID;
+import com.hedera.hapi.node.base.Key;
+import com.hedera.hapi.node.base.ScheduleID;
+import com.hedera.hapi.node.scheduled.ScheduleSignTransactionBody;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.service.contract.impl.exec.gas.DispatchType;
+import com.hedera.node.app.service.contract.impl.exec.gas.SystemContractGasCalculator;
 import com.hedera.node.app.service.contract.impl.exec.systemcontracts.common.AbstractCallTranslator;
 import com.hedera.node.app.service.contract.impl.exec.systemcontracts.common.Call;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hss.DispatchForResponseCodeHssCall;
 import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hss.HssCallAttempt;
 import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.ReturnTypes;
+import com.hedera.node.app.service.contract.impl.hevm.HederaWorldUpdater;
+import com.hedera.node.config.data.ContractsConfig;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Set;
+import javax.inject.Inject;
 import javax.inject.Singleton;
 
 /**
  * Translates {@code signSchedule()} calls to the HSS system contract.
  */
 @Singleton
 public class SignScheduleTranslator extends AbstractCallTranslator<HssCallAttempt> {
-    /** Selector for signSchedule(address,bytes) method. */
     public static final Function SIGN_SCHEDULE = new Function("signSchedule(address,bytes)", ReturnTypes.INT_64);
+    public static final Function SIGN_SCHEDULE_PROXY = new Function("signSchedule()", ReturnTypes.INT_64);
+    public static final Function AUTHORIZE_SCHEDULE = new Function("authorizeSchedule(address)", ReturnTypes.INT_64);
+
+    @Inject
+    public SignScheduleTranslator() {
+        // Dagger2
+    }
 
     @Override
     public boolean matches(@NonNull final HssCallAttempt attempt) {
         requireNonNull(attempt);
-        return attempt.isSelector(SIGN_SCHEDULE);
+        final var signScheduleEnabled =
+                attempt.configuration().getConfigData(ContractsConfig.class).systemContractSignScheduleEnabled();
+        final var authorizeScheduleEnabled =
+                attempt.configuration().getConfigData(ContractsConfig.class).systemContractAuthorizeScheduleEnabled();
+        return attempt.isSelectorIfConfigEnabled(signScheduleEnabled, SIGN_SCHEDULE_PROXY)
```

Comment:
why is this `SIGN_SCHEDULE_PROXY` and not `SIGN_SCHEDULE`?  (Or should it be both?)

Reply:
We are not support `SignSchedule` yet.  This is the version that is callable from a contract and takes a signature map